### PR TITLE
feat: experimental command line flag to print the dependency tree

### PR DESCRIPTION
**Pull request by kyegupov**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Experimental undocumented flag `--print-dependencies` for `snyk test`, useful for debugging.

image

### CODE REVIEW

File: src/lib/snyk-test/run-test.ts [hunk 1]
```diff
@@ -255,6 +266,9 @@ async function assembleLocalPayloads(root, options): Promise<Payload[]> {
 
     for (const depRoot of deps.depRoots) {
       const pkg = depRoot.depTree;
+      if (options['print-dependencies']) {
```
Comment (adrukh) [inline src/lib/snyk-test/run-test.ts]: Can I ask for `print-deps` instead? And have it appear in our help text? :pray:

Comment (kyegupov) [reply src/lib/snyk-test/run-test.ts]: Ok, changed it to be supported in both `test` and `monitor`.
I'd rather not make it a stable feature, so marked as "experimental" in help.
Probably it should be a separate command.

**Comment by miiila** [comment]
@kyegupov If it's not a big issue, I'd vote for a separate command as well. I believe that sorting out all the things like `snyk test --json --print-deps` would be too hard. If it is more complicated, I'm happy with experimental flag :-)

**Comment by miiila** [comment]
@kyegupov you handle `options.json` already... 👏

**Pull request merged**